scripts/qa_crowbarsetup.sh

```diff
@@ -3860,6 +3908,98 @@ function oncontroller_heat_image_setup()
     fi
 }
 
+function oncompute_parse_vm_xml
+{
+    local instance_name=$1
+    echo $instance_name
+    virsh list --all | grep $instance_name
+    [ $? != 0 ] && complain 53 "failed to find pci-passthru-instance"
+
+    # verify node xml to see if it has pci passthru section which has the
+    # specific address passing from compute host
+    virsh dumpxml $instance_name | grep -E "<hostdev" -A 7 | grep "0x1d"
+    [ $? != 0 ] && complain 63 "failed to find pci-passthru block in node xml"
+
+    return 0
+}
+
+function oncontroller_retrieve_image
```

Comment:
I was trying to reuse the `testvm` generated from `testsetup` step for past few days. However, I run into situation that the `testvm` won't boot up properly if the device passing from compute is attached (the specific flavor which has pci passthru attribute set). The workaround will be tweaking it to unplug/replug the device during post boot up phase. So I go back to my original formula instead of touching too many parts in `oncontroller_testsetup`.

Reply:
I understand your pain, but we need to think about this from a customer perspective. We should treat this like a bug that needs fixing instead of trying to work around it in our CI by using a different image (Cirros instead of SLES) ? However, I also understand that we don't really test the PCI passthrough feature in a manner that's relevant to our customers, so I'm a bit conflicted about this.

Can you give me more details about why the JeOS image won't boot with a second volume attached ? Is it because it tries to boot from the PCI passthrough device instead of the ephemeral volume set up by OpenStack ? If it's something we can fix easily through cloud-init, we should definitely try that. If not, I agree that we should use your original approach.